skimage/future/graph/graph_merge.py

```diff
@@ -128,10 +134,16 @@ def merge_hierarchical(labels, rag, thresh, rag_copy, in_place_merge,
             merge_func(rag, src, dst)
             new_id = rag.merge_nodes(src, dst, weight_func)
             _revalidate_node_edges(rag, new_id, edge_heap)
+            if merge_trace:
+                label_map = np.arange(labels.max() + 1)
+                for ix, (n, d) in enumerate(rag.nodes_iter(data=True)):
+                    for label in d['labels']:
+                        label_map[label] = ix
+                trace.append(label_map[labels])
```

Comment:
@jni Can we club `describe_func` and `weight_func` into 1 ? There is very little `weight_func` can do without something being already pre computed for it.